instance/ansible.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Ansible - Helper functions
+"""
+
+# Imports #####################################################################
+
+import os
+import subprocess
+import yaml
+
+from contextlib import contextmanager
+from tempfile import mkdtemp, mkstemp
+
+from django.conf import settings
+
+
+# Logging #####################################################################
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+# Functions ###################################################################
+
+def yaml_merge(yaml_str1, yaml_str2):
+    """
+    Merge the two yaml strings, recursively overriding variables from `yaml_str1` by `yaml_str2`
+    """
+    if not yaml_str2:
+        return yaml_str1
+
+    dict1 = yaml.load(yaml_str1)
+    dict2 = yaml.load(yaml_str2)
+    result_dict = dict_merge(dict1, dict2)
+
+    return yaml.dump(result_dict)
+
+
+def dict_merge(dict1, dict2):
+    """
+    Merge the two dicts, recursively overriding keys from `dict1` by `dict2`
+    """
+    result_dict = dict1.copy()
+    for key in dict2:
+        if key in result_dict and isinstance(result_dict[key], dict) and isinstance(dict2[key], dict):
+            result_dict[key] = dict_merge(result_dict[key], dict2[key])
```

Comment:
Nit: The above would read nicer to me if all instances of `result_dict` were replaced with `dict1` (except the instance where you assign to it, of course).